src/errors/ElementAlreadyDefinedError.ts

```diff
@@ -0,0 +1,5 @@
+export class ElementAreadyDefinedError extends Error {
```

Comment:
The name of this class has a typo which will affect places where it is imported as well.

```suggestion
export class ElementAlreadyDefinedError extends Error {
```